web/wp-content/plugins/tmbi-salesforce-dmp/class-tmbi-salesforce-dmp.php

```diff
@@ -0,0 +1,343 @@
+	const INTERCHANGE_SCRIPT_TAG = "<script class='kxint' data-namespace='trustedmediabrandsinc' type='text/javascript'>%s</script>";
+
+	/**
+	 * Variable to exclude the taxonomies
+	 *
+	 * @var array
+	 */
+	public static $excluded_taxonomies = array( 'yst_prominent_words', 'post_tag', 'category', 'exclude_feed', 'post_format', 'author' );
+
+	/**
+	 * Variable to build the krux data for localization
+	 *
+	 * @var array
+	 */
+	public static $krux_data_layer = array();
+
+	/**
+	 * Initiate the methods
+	 */
+	public static function init() {
+		add_action( 'wp_head', array( __CLASS__, 'render_salesforce_dmp_tags' ), self::PRIORITY );
+		add_action( 'wp_enqueue_scripts', array( __CLASS__, 'load_krux_click_tracking' ) );
+		add_action( 'wp_footer', array( __CLASS__, 'remove_krux_tracking' ), 1 );
+	}
+
+	/**
+	 * Localize the Krux targeting data
+	 */
+	public static function load_krux_click_tracking() {
+		wp_register_script(
+			self::SCRIPT_SLUG,
+			plugin_dir_url( __FILE__ ) . self::SCRIPT_URL,
+			array( 'jquery' ),
+			self::VERSION,
+			false
+		);
+		if ( class_exists( 'Ad_Stack' ) ) {
+			wp_localize_script(
+				'ad-stack',
+				self::SCRIPT_VAR,
+				self::krux_build_datalayer()
+			);
+		}
+	}
+
+	/**
+	 * Add the salesforce tags
+	 */
+	public static function render_salesforce_dmp_tags() {
+
+		$control_tag_script = PHP_EOL . "window.Krux||((Krux=function(){Krux.q.push(arguments)}).q=[]);
+  (function(){
+    var k=document.createElement('script');k.type='text/javascript';k.async=true;
+    k.src=(location.protocol==='https:'?'https:':'http:')+'//cdn.krxd.net/controltag/s9xpab5u5.js';
+    var s=document.getElementsByTagName('script')[0];s.parentNode.insertBefore(k,s);
+  }());" . PHP_EOL;
+
+		$contol_tag = sprintf( PHP_EOL . PHP_EOL . self::CONTROL_TAG_START . PHP_EOL . self::CONTROL_SCRIPT_TAG . PHP_EOL . self::CONTROL_TAG_END . PHP_EOL, $control_tag_script );
+
+		$interchage_tag_script = PHP_EOL . "window.Krux||((Krux=function(){Krux.q.push(arguments);}).q=[]);
+(function(){
+  function retrieve(n){
+    var k= 'kx'+'trustedmediabrandsinc_'+n, ls=(function(){
+      try {
+        return window.localStorage;
+      } catch(e) {
+        return null;
+      }
+    })();
+    if (ls) {
+        return ls[k] || '';
+    } else if (navigator.cookieEnabled) {
+        var m = document.cookie.match(k+'=([^;]*)');
+        return (m && unescape(m[1])) || '';
+    } else {
+        return '';
+    }
+  }
+  Krux.user = retrieve('user');
+  Krux.segments = retrieve('segs') ? retrieve('segs').split(',') : [];
+})();" . PHP_EOL;
+
+		$allowed_html   = [
+			'script' => [
+				'class'          => [],
+				'data-id'        => [],
+				'data-timing'    => [],
+				'data-version'   => [],
+				'type'           => [],
+				'data-namespace' => [],
+			],
+		];
+		$interchage_tag = sprintf( PHP_EOL . self::INTERCHANGE_SCRIPT_TAG, $interchage_tag_script );
+		$variant        = get_query_var( 'variant' );
+		if ( 'noads' !== $variant ) {
```

Comment:
After this is merged, we can integrate it with the Ad Blocker plugin, so we don't have to maintain that logic here.